### PR TITLE
Safely override `.kind` and `Buffer` improvements

### PR DESCRIPTION
**Pull request by daogrady**
Only add `override` modifier to `kind` property iff any parent class actually has such a property to override.
This change is implicitly tested by having added `strict: true` to the transpilation tests in `tscheck.js`.
Fixes https://github.com/cap-js/cds-typer/issues/273

Generally, the creation of class members has been streamlined to avoid typos and incorrect order/ combination of modifiers via `lib/components.class::createMember` instead of concatenating strings manually.

Also streamlined use of `Buffer.add` to accept scalar strings, array of strings, and functions returning strings, to enable `buffer.add(xs)` over `xs.forEach(x => buffer.add(x))`.
Also added `buffer.blankLine()` to replace the somewhat obscure `buffer.add('')`

### CODE REVIEW

File: lib/resolution/entity.js [hunk 1]
```diff
@@ -61,6 +61,7 @@ class EntityInfo {
     /** @type {import('../typedefs').resolver.EntityCSN | undefined} */
     #csn
 
+    /** @returns the **inferred** csn for this entity. */
```
Comment (chgeo) [inline lib/resolution/entity.js:64]: Do we have a type for an (inferred) CSN?  
If not, it should ideally come from `cds-types`. (Should do this in a separate PR though).

**Review by chgeo — APPROVED** [review]

File: lib/resolution/entity.js (hunk 1 above)
Comment (daogrady) [reply lib/resolution/entity.js:64]: We do not. The two packages were formerly strictly separated and CSN was only outlined. This could be added now that `cds-types` has become a (peer) dependency.

Comment (daogrady) [reply lib/resolution/entity.js:64]: I jotted down an issue https://github.com/cap-js/cds-typer/issues/368

Comment (chgeo) [reply lib/resolution/entity.js:64]: Yes.  `cds-types` should really become the goto-package for CDS types, not only in apps, but also in our tools.  We have more use cases in vscode-cds.
